### PR TITLE
Tbolt/2582 Allow additional affiliation requests

### LINKED ISSUE #2582: Create form for user to request additional states
Add an item to the dropdown menu that allows the user to request access to additional states. They should not have the ability to request access to a state they already have access to or remove access to a state they already have access to.

### This task is done when...
 - [ ] the user can click the Manage account item in the dropdown
 - [ ] clicking Manage account takes the user to the request access form

### PR DESCRIPTION
**Pull request by tbolt**
Resolves #2582 

**Description-**
This introduces a new page "Manage Account" that can be accessed from the top right nav. On this page the users existing affiliations are listed (currently ALL of them including requests/revoked/denied, we will likely update this). This page also allows a user to request access to additional states.

**This pull request changes...**

- Adds "Manage Account" link to top right navigation
- Allows viewing of existing affiliations
- Allows requesting additional affiliations

**Steps to manually verify this change...**

1. Login to the application with `norole` test account. Request access to 1(or more) states.
2. Navigate to "Manage Account" via newly added "Manage Account" link in the nav
3. Verify the previously requested state(s) appear under Current Affiliations
4. Add a new state to be requested
5. Submit request
6. Should see the "Thank You" message

### This pull request is ready to review when...

- [ ] Automated tests are updated (and all tests are passing)
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [x] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/containers/admin/ManageAccount.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  };
+
+  const handleCompleteAccessRequest = () => {
+    completeAccessRequest();
+    history.push('/');
+  };
+
+  return (
+    <div>
+      {showConfirmation && <StateAccessRequestConfirmation action={handleCompleteAccessRequest} />}
```
Comment (knollfear) [inline web/src/containers/admin/ManageAccount.js:34]: would a ternary operator be more clear here?

showConfirmation ?
 <StateAccessRequestConfirmation ....>
: 
 
<StateAccessRequest ...>

File: web/src/containers/StateAccessRequest.js [hunk 2]
```diff
@@ -1,16 +1,35 @@
...
-const StateAccessRequest = ({ saveAction, errorMessage, fetching }) => {
+const StateAccessRequest = ({ 
+  saveAction, 
+  errorMessage, 
+  fetching, 
+  currentAffiliations
+}) => {
+  // Todo: Exclude revoked/denied states?
+  const existingAffiliations = currentAffiliations.map(element => { 
+    const stateDetails = STATES.find(item => item.id === element.state_id)
```
[the remark this answers is not in the document]
Comment (tbolt) [reply web/src/containers/StateAccessRequest.js:19]: Yea, not a bad idea. I was re-using `/utils/states` which already existed and is used in other places. You think re-factoring it and updating it wherever it's used is a good option?

File: web/src/containers/StateAccessRequest.js [hunk 3]
```diff
@@ -1,16 +1,35 @@
...
+    return { id: element.state_id, name: stateDetails.name, disabled: true }
```
[the remark this answers is not in the document]
Comment (tbolt) [reply web/src/containers/StateAccessRequest.js:20]: nope, good catch

File: web/src/containers/StateAccessRequest.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (knollfear) [reply web/src/containers/StateAccessRequest.js:19]: maybe just create a second "STATEMAP" that is in that format would be easiest

**Review by knollfear — APPROVED** [review]
LGTM.  Nice work.

**Comment by eStar386** [comment]
Documented Functional Issue with the sorted list: When testing the ability to add states and eliminating selected states (seen in the current affiliations portion) from the sorted list. The list of states is returned to the user unsorted.
unsorted List

**Comment by eStar386** [comment]
Failed the accessibility testing for the confirmation page.

**Review by eStar386 — CHANGES_REQUESTED** [review]
This is ready for review once the confirmation page passes the wave. Passes functional and accessibility tests.

**Review by eStar386 — APPROVED** [review]
image (3)

Per a conversation with @tbolt there is a problem with the build and not with the actual confirmation page. The screenshot above was provided by Ty.

**Comment by eStar386** [comment]
Passes functional requirements of allowing a user to select multi-state affiliations and noted that the accessibility failures were due to a build issue.

**Comment by tbolt** [comment]
@eStar386 I did end up seeing that issue. I have since made an update to resolve it. Please verify you are not seeing it anymore. Thanks

**Review by eStar386 — APPROVED** [review]
Per the request of @tbolt , I reviewed the committed changes. Confirming it passes functional and accessible checks.  
Screen Shot 2021-05-28 at 10 28 05 AM

**Review by beparticular — APPROVED** [review]

**Review by NAretakis — APPROVED** [review]
Good to go!

**Pull request merged**